cmds/rm/rm.go

```diff
@@ -11,81 +11,166 @@
 //     -i: interactive mode
 //     -v: verbose mode
 //     -R: remove file hierarchies
-//     -r: equivalent to -R
+
 package main
 
 import (
 	"bufio"
+	"errors"
 	"flag"
 	"fmt"
+	"io/ioutil"
 	"os"
-	"path"
+	"path/filepath"
 )
 
-var (
-	flags struct {
-		r bool
-		v bool
-		i bool
+type rmFlags struct {
+	recursive   bool
+	verbose     bool
+	interactive bool
+}
+
+func isEmpty(file string) (bool, error, []os.FileInfo) {
+	fileList, err := ioutil.ReadDir(file)
+	if err != nil {
+		//ignore this boolean return value
+		return false, err, fileList
 	}
-	cmd = "rm [-Rrvi] file..."
-)
+	if len(fileList) == 0 {
+		return true, nil, fileList
+	}
+	return false, nil, fileList
+}
 
-func init() {
-	defUsage := flag.Usage
-	flag.Usage = func() {
-		os.Args[0] = cmd
-		defUsage()
+func interactivePrompt(input *bufio.Scanner, s string, args ...interface{}) (bool, error) {
+	for {
+		fmt.Printf(s+" (y/n)", args...)
+		input.Scan()
+		if input.Err() != nil {
+			return false, input.Err()
+		}
+		switch input.Text() {
+		case "y":
+			return true, nil
+		case "n":
+			return false, nil
+		}
+		fmt.Printf("Please only enter y or n. You entered %s. \n", input.Text())
 	}
-	flag.BoolVar(&flags.i, "i", false, "Interactive mode.")
-	flag.BoolVar(&flags.v, "v", false, "Verbose mode.")
-	flag.BoolVar(&flags.r, "R", false, "Remove file hierarchies")
-	flag.BoolVar(&flags.r, "r", false, "Equivalent to -R.")
-	flag.Parse()
 }
 
-func rm(files []string) error {
-	f := os.Remove
-	if flags.r {
-		f = os.RemoveAll
+func printRemove(flags rmFlags, input *bufio.Scanner, s1 string, s2 string, file string) (bool, error) {
```

Comment:
`s1` and `s2` are not very descriptive names. I'm wondering, though, if you couldn't pass the file's FileStat down here and have printRemove figure out what to print on its own based on the FIleStat. That way, all the remove strings would be in one place -- here.